### PR TITLE
feat(crux-ui): yup localization

### PR DESCRIPTION
**Pull request by robot9706**
- Localize yup schemas using labels.
- Add invalid instance to the toast when deployment validation fails.
- Add red ring around invalid input.
- Refactored config editor error messages a bit (for example environment messages didn't show up)

### CODE REVIEW

File: web/crux-ui/locales/en/common.json [hunk 1]
```diff
@@ -229,5 +229,9 @@
   },
 
   "imageConfig":"Image config",
-  "instanceConfig":"Container config"
+  "instanceConfig":"Container config",
+
+  "validation": {
```
Comment (m8vago) [inline web/crux-ui/locales/en/common.json:234]: Please user `error.json`

File: web/crux-ui/src/components/projects/versions/images/config/common-config-section.tsx [hunk 2]
```diff
@@ -167,7 +165,7 @@ const CommonConfigSection = (props: CommonConfigSectionProps) => {
                   placeholder={t('common.containerName')}
                   onPatch={it => onChange({ name: it })}
                   editorOptions={editorOptions}
-                  message={fieldErrors.find(it => it.path?.startsWith('name'))?.message}
+                  message={findErrorFor(fieldErrors, 'name')}
```
Comment (m8vago) [inline web/crux-ui/src/components/projects/versions/images/config/common-config-section.tsx:168]: :heart:

File: web/crux-ui/src/hooks/use-dyo-formik.ts [hunk 3]
```diff
@@ -1,17 +1,39 @@
...
+      ? () => ({
+          ...options.validationSchema,
+          validateSync: (values: any, validationOptions: any) => {
+            try {
+              return options.validationSchema.validateSync(values, validationOptions)
+            } catch (err) {
+              throw t ? yupErrorTranslate(err, t) : err
+            }
+          },
+          validate: (values: any, validationOptions: any) =>
```
Comment (m8vago) [inline web/crux-ui/src/hooks/use-dyo-formik.ts:28]: I believe you can get rid of the `new Promise` with declaring this as `async` and a `try catch`.

File: web/crux-ui/src/validations/image.ts [hunk 4]
```diff
@@ -1,9 +1,61 @@
...
 import { getValidationError } from './common'
 import { containerConfigSchema } from './container'
+import { Translate } from 'next-translate'
 
-export const getContainerConfigFieldErrors = (newConfig: ContainerConfigData): ValidationError[] =>
-  getValidationError(containerConfigSchema, newConfig, { abortEarly: false })?.inner ?? []
+export type ContainerConfigValidationErrors = Record<string, yup.ValidationError>
 
-export const jsonErrorOf = (fieldErrors: ValidationError[]) => (fieldErrors.length > 0 ? fieldErrors[0].message : null)
+export type ErrorWithPath = {
```
Comment (m8vago) [inline web/crux-ui/src/validations/image.ts:9]: I think this should be in `common.ts`

File: web/crux-ui/src/validations/yup.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+const createMessage = (original: Message) => (params: any) => {
+  const newParams = {
+    ...params,
+    path: params.label ? `\${${params.label}}` : params.path,
+    regex: params.spec?.meta?.regex ? `\${${params.spec?.meta?.regex}}` : params.regex,
+  }
+
+  if (typeof original === 'string') {
+    return original.replace(/\$\{\s*((\w|:|\.)+)\s*\}/g, (_, key) => yup.printValue(newParams[key]))
```
Comment (m8vago) [inline web/crux-ui/src/validations/yup.ts:13]: Please write some explanations for regexps in a comment.

**Review by m8vago — CHANGES_REQUESTED** [review]

**Review by m8vago — CHANGES_REQUESTED** [review]

File: web/crux-ui/locales/en/container.json [hunk 6]
```diff
@@ -134,5 +138,12 @@
       "metricsPath": "/metrics"
     }
   },
-  "sensitiveKey": "Sensitive config data should be stored as secrets"
+
+  "validation": {
+    "kubernetesQuantity": "{{path}} must be a valid Kubernetes quantity",
+    "noSensitive": "{{path}} must not contain sensitive keywords",
+    "pathOverlapsSomePortranges": "{{path}} overlaps some portranges",
```
Comment (polaroi8d) [inline web/crux-ui/locales/en/container.json:145]: "port ranges"

**Review by polaroi8d — APPROVED** [review]
Wow, good job. LGTM 🦄

**Review by m8vago — APPROVED** [review]

**Pull request merged**